### PR TITLE
PICARD-204: Support for recording-level original release date in scripting

### PR DESCRIPTION
**Pull request by phw**
# Summary

* This is a…
    * [ ] Bug fix
    * [x] Feature addition
    * [ ] Refactoring
    * [ ] Minor / simple change (like a typo)
    * [ ] Other
* **Describe this change in 1-2 sentences**:

# Problem

* JIRA ticket (_optional_): PICARD-204

This is an alternative implementation of https://github.com/metabrainz/picard/pull/1732

# Solution

Provide variables `%_recording_firstreleasedate%` and `%_releasegroup_firstreleasedate%` and set `originaldate` / `originalyear` for non-album tracks. For releases the tag variables for `originaldate` and `originalyear` will still be set from the release group's first release date, but the user can choose to use the recording date instead.

E.g. things like this are possible:

```
$set(originaldate,$if2(%_recording_firstreleasedate%,%_releasegroup_firstreleasedate%))
```

```
$set(originaldate,$if($inmulti(%releasetype%,compilation),%_recording_firstreleasedate%,%_releasegroup_firstreleasedate%))
```

# Action

### CODE REVIEW

**Comment by phw** [comment]
See discussion at https://github.com/metabrainz/picard/pull/1732

File: test/test_mbjson.py [hunk 1]
```diff
@@ -54,7 +54,8 @@
     "standardize_instruments": True,
     "release_ars": True,
     "preferred_release_countries": [],
-    "artist_locale": 'en'
+    "artist_locale": 'en',
+    "originaldate_use_recording": True
```
Comment (zas) [inline test/test_mbjson.py:58]: Since Python allows it, we should keep a comma at the end of the last line, it makes diffs cleaner.

**Review by zas — CHANGES_REQUESTED** [review]
Apart minor comment about comma, LGTM.

File: test/test_mbjson.py (hunk 1 above)
Comment (phw) [reply test/test_mbjson.py]: Actually this option was removed, so this line is not needed at all. I added the comma anyway as a small syntax change, we usually keep trailing commas in those cases. I would be even ok if we would test for this in our linting

**Pull request merged**